tests/test_gruptree.py

```diff
@@ -272,8 +380,67 @@ def test_main(tmpdir):
 
 def test_prettyprint():
     """Test pretty printing via command line interface"""
-    sys.argv = ["ecl2csv", "gruptree", DATAFILE, "--prettyprint"]
-    ecl2csv.main()
+    commands = ["ecl2csv", "gruptree", DATAFILE, "--prettyprint"]
+    results = subprocess.run(commands, check=True, stdout=subprocess.PIPE)
```

Comment:
I don't think it is wise to run tests in subprocess, it just adds an additional layer making debugging more difficult. Previous implementation modified `sys.argv`, which is also not a very nice thing to do, so I understand the reason to change. I would suggest that instead the test mocks `sys.argv` before calling `main`

Reply:
Ok, unaware of that (but I have a hunch someone actually recommended `subprocess.run` in tests elsewhere), but have been surprised recently that the coverage reports seemed to be able to see through `subprocess.run` .

Reply:
Well, I would I least recommend mocking sys argv instead. Modifying it during runtime is worse, as you are actually modifying it for every test (so if you were to append anything to it in every test, you would end up with a very long sys.argv..). Sometimes you probably want to use subprocess, for instance if you want to verify that the entry point is correctly installed. I would reserve those tests to the simple scenario of `subprocess.run(["name", -h"])`

Reply:
Ok, thanks. Will use the opportunity to try `pytest-mock`

Reply:
Rebased, and a new commit added changing to mocking.